packages/core/upload/admin/src/index.js

```diff
@@ -14,72 +14,70 @@ import Initializer from './containers/Initializer';
 import SettingsPage from './containers/SettingsPage';
 import InputMedia from './components/InputMedia';
 import InputModalStepper from './containers/InputModalStepper';
-
+import reducers from './reducers';
 import trads from './translations';
 import pluginId from './pluginId';
 import { getTrad } from './utils';
 
-export default strapi => {
-  const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
-  const icon = pluginPkg.strapi.icon;
-  const name = pluginPkg.strapi.name;
+const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
+const icon = pluginPkg.strapi.icon;
+const name = pluginPkg.strapi.name;
+
+export default {
+  register(app) {
+    app.componentApi.registerComponent({ name: 'media-library', Component: InputModalStepper });
+    app.fieldApi.registerField({ type: 'media', Component: InputMedia });
```

Comment:
I completely agree, this is why I have planned to rework these APIs in another PR. I thought that it would be easier to create smaller PRs so it is easier to spot the differences.